samples/golang/webfrontend/src/app/main.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"html/template"
+	"log"
+	"net"
+	"net/http"
+	"strconv"
+)
+
+// serverMessage is a representation of a message from server
+type serverMessage struct {
+	Value string
+}
+
+// pageVariables is a representation of a server page
+type pageVariables struct {
+	PageTitle   string
+	PageMessage serverMessage
+	PageButton  string
+}
+
+var counter int
+
+func main() {
+	http.HandleFunc("/", HomePage)
+	http.HandleFunc("/api", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "Hello from webfrontend")
+	})
+	listener, err := net.Listen("tcp", ":80")
+	if err != nil {
+		log.Fatal("Listen failed: ", err)
+	}
+	fmt.Println("Server is listening on port", listener.Addr().(*net.TCPAddr).Port)
+	log.Fatal(http.Serve(listener, nil))
+}
+
+// HomePage is triggered on load of the server
+func HomePage(w http.ResponseWriter, r *http.Request) {
+	title := "Server"
+	myButton := "Click Me (Reload)"
+	messages := make(chan string)
+	go func() {
+		counter++
```

Comment:
When I test this, the counter starts off at 1 and then increments by **2** each time I hit reload? Not sure if there is anything you can do about it in code (obviously the code only increments by 1 each time, perhaps it's the browser that's calling the method twice?) but I thought I'd mention it in case there is.

Reply:
I can see what you are saying. I remember running it before and it worked fine then. I am looking into it.